std/datetime.d

```diff
@@ -1075,6 +1075,25 @@ public:
         //assert(ist.opCmp(ist) == 0);
     }
 
+    /**
+     * Returns: A hash of the $(LREF SysTime)
+     */
+    size_t toHash() const @nogc pure nothrow @safe
+    {
+        if (_stdTime < 0)
+            return (cast(size_t) (_stdTime * -1)) + long.max;
```

Comment:
> I'm not an expert in hashing, but really this is going to be truncated down to mod of the table size. So why is it important to use a complex function for this? I thought hashing small values was a waste of CPU.

No. I mean only finalization step of MurmurHash2, not algorithms itself. It is very fast.

> You may want to add upper 32-bits to lower 32-bits in any case (on 32-bit version), since lower 32-bits is likely to be very repetitive and more reflective of the clock resolution.

Is it true for truncated time values, for example truncated to year?